include/circt/Support/PrettyPrinter.h

```diff
@@ -72,14 +72,24 @@ class Token {
   struct EndInfo : public TokenInfo {
     // Nothing
   };
+  // This can be used to associate a callback with the print event on the
+  // tokens stream. The id, is used to uniquely identify an intereseting event
+  // on the token stream.
+  struct CallbackInfo : public TokenInfo {
+    using CallbackTy = llvm::function_ref<void(unsigned)>;
+    CallbackTy callback;
+    unsigned id;
```

Comment:
```suggestion
    uint32_t id;
```
(or `uint64_t` if continue to make these 32 bytes anyway).